### ISSUE TITLE
Share button should appear in the editor for logged out users

### ISSUE THREAD

**Issue by paulkaplan** [LLK/scratch-www#4394]
Currently the share button does not appear for logged out users. https://github.com/LLK/scratch-www/issues/4393 addresses the same issue for unconfirmed email users who also currently can't share. But the "Share" button should appear for logged out users as well, and it should bring up a modal prompting "Join or Sign In" 
<img width="666" alt="Screen Shot 2020-09-02 at 3 09 05 PM" src="https://user-images.githubusercontent.com/654102/92025978-49c2c080-ed2e-11ea-9655-a291a6a2d647.png">

Join should bring up the existing join flow modal, and sign in will bring up a new sign in modal (which we do not have yet), roughly this without the red pip (sorry)

<img width="428" alt="Screen Shot 2020-09-02 at 3 11 23 PM" src="https://user-images.githubusercontent.com/654102/92026196-a7570d00-ed2e-11ea-80cc-1a1e82abaf2c.png">

@carljbowman I just want to confirm that clicking share and then clicking "sign in" or "join", going through those flows, is the outcome that the project is actually shared? Just want to make sure, submitting that sign in form will both sign you in _and_ share, or will it just sign you in, and you repeat clicking "share".

**Comment by paulkaplan** [comment]
Also note, part of this should also be making the project title editable while logged out. This should allow people to rename a project in order to save it offline, as well as keep the nav bar more visually consistent between logged out and in.

**Comment by carljbowman** [comment]
**Re functionality after completing signing-in or joining:**
That's a good question to clarify. My gut instinct would be to not combine the actions and have the project become shared after signing in or joining. Instead have the user click Share again after having signed in or joined to take that step. 

I have some concern about mixing the two actions and the focus on project share state and login state. While the proposal above requires an additional redundant step, to me, that redundancy is helping clarify the shared state of a project.

Additionally, I think in the future we may want to revisit the share flow in the future to...
1) Give more feedback that a project has been shared (the editor just transitions the button, where as the project page banner on the project page shows a "Congrats" message).
2) Provide a flow to fill out project details (e.g. name, instructions, notes and credits...) while providing some the feedback mentioned in (1). Also, seems like a good opportunity to welcome new comers into the community if it's the first time sharing. 

The reason I mention that the future share flow stuff is that I think by coupling the login/join + share action may add even more navigation and state complexity being handed between those flows.

That said, certainly open to other thoughts or opinions on this though, as I believe I had been implicitly thinking these actions would not be coupled and it would be good to be explicit.

**Re assets:**
Here is a first pass at handing-off some assets. As mentioned on https://github.com/LLK/scratch-www/issues/4390, I kept the illustration separate from the green blob background, as that blob background is used on this modal and the linked to modal.

That said, I am happy to combine assets and/or format these in anyway. Let me know if you have preferences.

[join or sign in modal assets.zip](https://github.com/LLK/scratch-www/files/5169170/join.or.sign.in.modal.assets.zip)

**Comment by paulkaplan** [comment]
Yeah agree that it probably should just put you back into the editor, signed in, where you can try sharing again. Probably wouldn't be good to have share happen directly from join!

**Comment by carljbowman** [comment]
As per the asset request in https://github.com/LLK/scratch-www/issues/4390, here is a new version of the asset.

[join-signin-illustration.svg.zip](https://github.com/LLK/scratch-www/files/5218858/join-signin-illustration.svg.zip)
